src/System.IO.Pipes/tests/NamedPipesThrowsTests.cs

```diff
@@ -536,15 +833,34 @@ public static void ClientNotConnectedThrows()
 
                     server.Dispose();
 
-                    Assert.Throws<IOException>(() => client.Write(buffer, 0, buffer.Length));
-                    Assert.Throws<IOException>(() => client.WriteByte(123));
-                    Assert.Throws<IOException>(() => client.Flush());
-                    int length = client.Read(buffer, 0, buffer.Length);
-                    Assert.Equal(0, length);
-                    int byt = client.ReadByte();
+                    OtherSidePipeDisconnectWriteThrows(client);
+                    OtherSidePipeDisconnectVerifyRead(client);
                 }
             }
+        }
+
+        [Fact]
+        public static void ClientConnectWrongConflictingDirection()
+        {
+            const string serverName1 = "testServer4";
+
+            using (NamedPipeServerStream server = new NamedPipeServerStream(serverName1, PipeDirection.Out))
+            using (NamedPipeClientStream client = new NamedPipeClientStream(".", serverName1, PipeDirection.Out))
+            {
+                Assert.Throws<UnauthorizedAccessException>(() => client.Connect());
```

Comment:
And the hang on Linux is coming from here.  As it's valid to have multiple writers open a FIFO, there's no error associated with this, as the client in this case ends up looking like any other writer trying to open the fifo, blocking until a reader connects.  Not sure of a solution at the moment; there might be another underlying mechanism for NamedPipe*Streams that would make this work correctly, but I fear it would lead to similar differences elsewhere.  Thoughts?